app/Models/Bible/BibleBook.php

```diff
@@ -119,4 +138,51 @@ public function book()
     {
         return $this->belongsTo(Book::class);
     }
+
+    /**
+     * Get collection of bible books sorted by bible versification or the book_seq column if it not empty.
+     *
+     * @param string $bible_id
+     * @param string $book_id
+     * @param string $bible_versification
+     *
+     * @return Collection
+     */
+    public static function getAllSortedByBookSeqOrVersification(
+        string $bible_id,
+        ?string $book_id = null,
+        ?string $bible_versification = null
+    ) : Collection {
+        $select_columns = [
+            'bible_books.bible_id',
+            'bible_books.book_id',
+            'bible_books.name',
+            'bible_books.name_short',
+            'bible_books.chapters',
+            'bible_books.book_seq'
+        ];
+
+        if ($bible_versification) {
```

Comment:
since bible_versification is non-null, take off this if check. It is confusing

Reply:
@bradflood , this validation has been took off.